tensorboard/uploader/uploader_main.py

```diff
@@ -733,9 +743,13 @@ def _get_intent(flags):
 
 def _get_server_info(flags):
```

Comment:
Changes to fix the issues in pr 3344 are all in this function.

Happy to volunteer to improve test coverage for some of the code in this file. Would it be appropriate to create an Issue for that?

Reply:
Yes, feel free to open an issue that `uploader_main.py` needs tests.
This goes hand in hand with turning `uploader_main.py` into more of a
library so that we can have a proper Python API, like

```python
import tensorboard

exp = tensorboard.dev.upload(logdir)
print(exp.url)
model.fit()
...
tensorboard.dev.delete(exp.experiment_id)
```

or something. It’s not obvious to me what the best way to test it is, so
opening an issue sounds good. thanks!

Reply:
Done. https://github.com/tensorflow/tensorboard/issues/3404